components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,634 @@
+	--parse opponents + set base sumscores
+	match = CustomMatchGroupInput._opponentInput(match)
+
+	--main processing done here
+	local subGroupIndex = 0
+	for _, _, mapIndex in Table.iter.pairsByPrefix(match, 'map') do
+		match, subGroupIndex = CustomMatchGroupInput._mapInput(match, mapIndex, subGroupIndex)
+	end
+
+	match = CustomMatchGroupInput._matchWinnerProcessing(match)
+
+	CustomMatchGroupInput._setPlacements(match)
+
+	if CustomMatchGroupInput._hasTeamOpponent(match) then
+		match = CustomMatchGroupInput._subMatchStructure(match)
+	end
+
+	if Logic.isNumeric(match.winner) then
+		match.finished = true
+	end
+
+	return match
+end
+
+---@param match table
+---@return table
+function CustomMatchGroupInput._matchWinnerProcessing(match)
+	local bestof = tonumber(match.bestof) or Variables.varDefault('bestof', DEFAULT_BEST_OF)
+	match.bestof = bestof
+	Variables.varDefine('bestof', bestof)
+
+	local scores = Array.map(Array.range(1, MAX_NUM_OPPONENTS), function(opponentIndex)
+		local opponent = match['opponent' .. opponentIndex]
+		if not opponent then
+			return NO_SCORE
+		end
+
+		-- set the score either from manual input or sumscore
+		opponent.score = Table.includes(ALLOWED_STATUSES, string.upper(opponent.score or ''))
+			and string.upper(opponent.score)
+			or tonumber(opponent.score) or tonumber(opponent.sumscore) or NO_SCORE
+
+		return opponent.score
+	end)
+
+	walkoverProcessing.walkover(match, scores)
+
+	if match.resulttype == DEFAULT_WIN_RESULTTYPE then
+		walkoverProcessing.applyMatchWalkoverToOpponents(match)
+		return match
+	end
+
+	if match.winner == 'draw' then
+		match.winner = 0
+	end
+
+	Array.forEach(Array.range(1, MAX_NUM_OPPONENTS), function(opponentIndex)
+		local opponent = match['opponent' .. opponentIndex]
+		if Logic.isEmpty(opponent) then return end
+		opponent.status = SCORE_STATUS
+		if opponent.score > bestof / 2 then
+			match.finished = Logic.emptyOr(match.finished, true)
+			match.winner = tonumber(match.winner) or opponentIndex
+		elseif match.winner == 0 or (opponent.score == bestof / 2 and match.opponent1.score == match.opponent2.score) then
+			match.finished = Logic.emptyOr(Logic.readBoolOrNil(match.finished), true)
+			match.winner = 0
+			match.resulttype = 'draw'
+		end
+	end)
+
+	match.winner = tonumber(match.winner)
+
+	CustomMatchGroupInput._checkFinished(match)
+
+	if match.finished and not match.winner then
+		CustomMatchGroupInput._determineWinnerIfMissing(match, scores)
+	end
+
+	return match
+end
+
+---@param match table
+function CustomMatchGroupInput._checkFinished(match)
+	if Logic.readBoolOrNil(match.finished) == false then
+		match.finished = false
+	elseif Logic.readBool(match.finished) or match.winner then
+		match.finished = true
+	end
+
+	-- Match is automatically marked finished upon page edit after a
+	-- certain amount of time (depending on whether the date is exact)
+	if not match.finished and match.timestamp > DateExt.defaultTimestamp then
```

Comment:
Is this intended logic, that if `match.finished=false` is set, it will still set it to finished after 10/24h?